packages/astro/src/runtime.ts

```diff
@@ -193,7 +203,7 @@ async function load(config: RuntimeConfig, rawPathname: string | undefined): Pro
         };
       }
 
-      collection.data = data;
+      collection.data = Array.isArray(data) ? data : [data];
```

Comment:
Another UX improvement: if the user didn’t return an array of data, convert it into one for them. While there’s probably other things going on, this should actually make their own errors clearer (and IMO this is better than erring out if the user didn’t return an array)